src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsnonassociationsapi/service/SubjectAccessRequestService.kt

```diff
@@ -18,14 +20,25 @@ class SubjectAccessRequestService(
     toDate: LocalDate?,
   ): HmppsSubjectAccessRequestContent? {
     val options = NonAssociationListOptions(includeOtherPrisons = true, inclusion = NonAssociationListInclusion.ALL)
-    val nonAssociations = nonAssociationsService.getPrisonerNonAssociations(prn, options)
 
-    return HmppsSubjectAccessRequestContent(
-      content = nonAssociations.copy(
-        nonAssociations = nonAssociations.nonAssociations.filter {
-          (fromDate == null || it.whenCreated.toLocalDate().isAfter(fromDate)) && (toDate == null || it.whenCreated.toLocalDate().isBefore(toDate))
-        },
-      ),
+    val nonAssociations = try {
+      nonAssociationsService.getPrisonerNonAssociations(prn, options)
+    } catch (e: NotFound) {
```

Comment:
This doesn't look right to me,  the nonAssociationsService.getPrisonerNonAssociations() has this slightly odd bit:- 
```{kotlin}
 // Throw an exception if any of the prisoners searched were not found
    val missingPrisoners = prisonerNumbers.subtract(foundPrisoners.keys)
    if (missingPrisoners.any()) {
      throw ResponseStatusException(
        HttpStatus.NOT_FOUND,
        "Could not find the following prisoners: $missingPrisoners",
      )
    }
```

Actually this should throw an explicit exception for this use case and the code below would catch that: e..g

```{kotlin}
if (missingPrisoners.any()) {
      throw MissingPrisonersInSearchException( // This is a new exception
        "Could not find the following prisoners: $missingPrisoners",
      )
    }
```
then in this class 
```{kotlin}
// Don't need try catch as exception will bubble up
      nonAssociationsService.getPrisonerNonAssociations(prn, options)
```

Then in Exception Handler 
```{kotlin}
 @ExceptionHandler(MissingPrisonersInSearchException::class)
  fun handleMissingPrisonersInSearchException(e: MissingPrisonersInSearchException): ResponseEntity<ErrorResponse?>? {
    log.debug("Missing prisoners in search: {}", e.message)
    return ResponseEntity
      .status(NOT_FOUND)
      .body(
        ErrorResponse(
          status = NOT_FOUND,
          userMessage = "Missing prisoners: ${e.message}",
          developerMessage = e.message,
        ),
      )
  }
```